org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/commands/XMLValidationFileCommand.java

```diff
@@ -52,12 +57,16 @@ public XMLValidationFileCommand(ContentModelManager contentModelManager, IXMLDoc
 	@Override
 	protected Object executeCommand(DOMDocument document, ExecuteCommandParams params, SharedSettings sharedSettings,
 			CancelChecker cancelChecker) throws Exception {
+		// Validation args can contains the external resource ('url' as key map) to
+		// force do download.
+		JsonObject validationArgs = params.getArguments().size() > 1 ? (JsonObject) params.getArguments().get(1) : null;
 		// 1. remove the referenced grammar in the XML file from the Xerces grammar pool
 		// (used by the Xerces validation) and the content model documents cache (used
 		// by the XML completion/hover based on the grammar)
 		contentModelManager.evictCacheFor(document);
 		// 2. trigger the validation for the given XML file
-		validationService.validate(document);
+		Map map = JSONUtility.toModel(validationArgs, Map.class);

```

Comment:
what happens when validationArgs == null?

Reply:
No force download url will be done when validation will occurs. The null arguments is managed at https://github.com/eclipse/lemminx/pull/1183/files#diff-07b75b69658ecc851282ac49a9a36f30a26fcb8e26004e6705d279dc094af2ebR91